chip8/parsing.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <memory>
+
+#include "types.h"
+#include "reg.h"
+#include "instruction.h"
+
+namespace parsing
+{
+	std::unique_ptr<instruction> parse_instruction(const word opcode);
```

Comment:
This is an heap allocation for each instruction. Consider profiling. Consider adding a flyweight/cache.